File: LES_SERIE_S1_IA_FS_ALGO_PYTHON_C/Série_de_TD-TP_4/exercice9.py
T=[[None for j in range(15)] for i in range(20)]
somme=0
elements_s_positif=0
somP=0
somN=0
for i in range(0,20):
    for j in range(0,15):
        T[i][j]=float(input(f"saisir l'élément de case {i+1} {j+1}: "))
        somme=somme+T[i][j]
        if T[i][j]>0:
            elements_s_positif+=1
            somP=somP+T[i][j]
        else:
            somN=somN+T[i][j]
max=T[0][0]
min=T[0][0]
for i in range(1,20):
    for j in range(1,15):
        if T[i][j]>max:
            max=T[i][j]
            pos_max=f"{i} {j}"
        if T[i][j]<min:
            min=T[i][j]
            pos_min=f"{i} {j}"
print(f"la somme de tous les éléments de la matrice est: {somme}")
print(f"le nombre des éléments strictement positifs est: {elements_s_positif}")
print(f"la somme des éléments positifs (SomP) est: {somP}")
print(f"la somme des éléments négatifs(SomN): {somN}")
print(f"le plus grand élément de la matrice {max} ainsi que sa position est {pos_max}")
print(f"le plus petit élément de la matrice {min} ainsi que sa position est {pos min}")

File: LES_SERIE_S1_IA_FS_ALGO_PYTHON_C/Série_de_TD-TP_4/exercice9.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
int N_total_elements_S_positif,max_pos[2],min_pos[2];
float somme,somP,somN,max,min,M[20][15];
int main(){
    N_total_elements_S_positif=0;
    somme=0;
    somP=0;
    somN=0;
    for(int i=0;i<20;i++){
        for(int j=0;j<15;j++){
            printf("saisir l'element de cas {%d , %d} : ",i+1,j+1);
            scanf("%f",&M[i][j]);
            somme+=M[i][j];
            if(M[i][j]>0){
                N_total_elements_S_positif++;
                somP+=M[i][j];
            }
            else{
                somN+=M[i][j];
            }
        }
    }
    max=M[0][0];
    min=M[0][0];
        somN=0;
    for(int i=0;i<20;i++){
        for(int j=0;j<15;j++){
            if(M[i][j]>max){
                max=M[i][j];
                max_pos[0]=i;
                max_pos[1]=j;
            }
            if(M[i][j]<min){
                min=M[i][j];
                min_pos[0]=i;
                min_pos[1]=j;
            }
        }
    }
    printf("la somme de tous les éléments de la matrice est: %f\n",somme);
    printf("le nombre des éléments strictement positifs est: %d\n",N_total_elements_S_positif);
    printf("la somme des éléments positifs (SomP) est: %f\n",somP);
    printf("la somme des éléments négatifs(SomN): %f\n",somN);
    printf("le plus grand élément de la matrice %f ainsi que sa position est {%d,%d}\n",max,max_pos[0],max_pos[1]);
    printf("le plus petit élément de la matrice %f ainsi que sa position est {%d,%d}\n",min,min_pos[0],min_pos[1]);
    return 0;
}